File: app/src/lib/modes.ts
export default new Map<string, { tag: string; title: string; name: string; description: string }>([
	[
		'classic-casual',
		{
			tag: 'classic',
			title: 'Casual',
			name: 'classic-casual',
			description: 'Just a normal, casual multiplayer snake'
		}
	],
	[
		'classic-huuge',
		{
			tag: 'classic',
			title: 'Huuge',
			name: 'classic-huuge',
			description: 'Quite big map with many players and apples'
		}
	],
	[
		'battleroyale-eat&win',
		{
			tag: 'battle-royale',
			title: 'Eat&Win',
			name: 'battleroyale-eat&win',
			description: 'Popular Battle Royale! Compete with friends, escape from zone, eat apples and win!'
		}
	],
	[
		'battleroyale-zoneshark',
		{
			tag: 'battle-royale',
			title: 'Zone Shark',
			name: 'battleroyale-zoneshark',
			description: "Don't touch the zone! It eats you instantly."
		}
	],
]);

File: server/pkg/core/rooms/modes.go
package rooms

import (
	"snakeish/pkg/core/rooms/battleroyale"
	"snakeish/pkg/core/rooms/classic"
)

func CreateMode(tag string, name string) Mode {
	switch tag {
	default: //classic
		switch name {
		default: //casual
			return &classic.Mode{
				ApplesQuantity: 3,
				GridSize:       8,
				MaxPlayers:     4,
				FrameTime:      250,
				ModeName:       "Casual",
			}
		case "Huuge":
			return &classic.Mode{
				ApplesQuantity: 8,
				GridSize:       16,
				MaxPlayers:     8,
				FrameTime:      250,
				ModeName:       "Huuge",
			}
		}

	case "battle-royale":
		switch name {
		default: //Eat&Win
			return &battleroyale.Mode{
				MaxPlayers:     8,
				MinPlayers:     3,
				FrameTime:      250,
				GridSize:       20,
				ShrinkSize:     0,
				ZoneShrinkTime: 15,
				ZoneKillTime:   3,
				AppleSpawnTime: 5,
				ModeName:       "Eat&Win",
			}
		case "Zone Shark":
			return &battleroyale.Mode{
				MaxPlayers:     8,
				MinPlayers:     3,
				FrameTime:      200,
				GridSize:       25,
				ShrinkSize:     0,
				ZoneShrinkTime: 10,
				ZoneKillTime:   0,
				AppleSpawnTime: 4,
				ModeName:       "Zone Shark",
			}
		}
	}
}
